packages/babel-generator/src/generators/base.ts

```diff
@@ -36,15 +36,11 @@ export function BlockStatement(this: Printer, node: t.BlockStatement) {
     this.printSequence(node.body, node, { indent: true });
     this.removeTrailingNewline();
 
-    this.sourceWithOffset("end", node.loc, 0, -1);
-
     if (!this.endsWith(charCodes.lineFeed)) this.newline();
```

Comment:
Can you add a test case for this? [This](https://astexplorer.net/#/gist/0d6d44fa70de0896139b41c9f365075c/52964b48370fd5f2b930bb0fdcee240bb0bf8afa) should work as an input.